src/dbnode/storage/database.go

```diff
@@ -437,7 +437,15 @@ func (d *db) AssignShardSet(shardSet sharding.ShardSet) {
 		ns.AssignShardSet(shardSet)
 	}
 
-	d.queueBootstrapWithLock()
+	if receivedNewShards {
```

Comment:
Can we add a test for this if we're adding this functionality?

Reply:
I think correctness is covered well by the existing integration tests but I can add a unit test to make sure we only bootstrap when receiving new shards